### PR TITLE
Feature/npl suggestions api

### PR DESCRIPTION
**Pull request by yneves**
* Integrated suggestions API done here https://github.com/syncids/syncids-api/pull/12.
 * Some refactoring to split `Documents` and `DocumentsContainer`.

### CODE REVIEW

File: src/components/documents/DocumentDescription.js [hunk 1]
```diff
@@ -111,6 +106,14 @@ export default class DocumentDescription extends Component {
     );
   }
 
+  shouldRenderSuggestions(value) {
+    if (!value || value.length < 3 || value === this.props.doc.get(this.props.field)) {
+      return false;
+    }
+    this.props.onSearch(value);
```
Comment (yneves) [inline src/components/documents/DocumentDescription.js]: This where the search is triggered.

File: src/components/documents/DocumentsContainer.js [hunk 2]
```diff
@@ -35,37 +26,20 @@ function mapStateToProps({app: {selectedMatterId, selectedClientId}, entities, d
...
-    addDocumentToMatter: PropTypes.func,
-    deleteDocumentFromMatter: PropTypes.func,
-    deleteDocumentFromClient: PropTypes.func,
-    uploadFile: PropTypes.func,
-    updateDocumentCitedState: PropTypes.func,
-    replaceDocNumber: PropTypes.func,
-    sortDocuments: PropTypes.func,
-    clearNewDocument: PropTypes.bool,
-    isLoadingDocuments: PropTypes.bool
+  const nplSuggestions = entities.getIn(['nplSuggestions', matter.get('clientNumber')]) || Immutable.Map();
```
Comment (yneves) [inline src/components/documents/DocumentsContainer.js]: Suggestions are cached just like documents.

File: src/redux/actions/document/searchNplDocuments.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+const searchNplDocuments = createAction(SEARCH_NPL_DOCUMENTS);
+
+export default (firmName, clientNumber, nonPatentBib) => (dispatch, getState) => {
+  if (!firmName || !clientNumber || !nonPatentBib) {
+    return;
+  }
+
+  const {documents: {searchNpl}} = getState();
+  if (searchNpl && searchNpl[clientNumber] && searchNpl[clientNumber][nonPatentBib]) {
```
Comment (yneves) [inline src/redux/actions/document/searchNplDocuments.js]: Store search terms for each client number so it doesn't call API again for the same values.

**Comment by yneves** [comment]
@ntgn81 this is ready!

**Comment by ntgn81** [comment]
Boo, conflicts!

**Pull request merged**